AsyncDisplayKit/ASVideoNode.mm

```diff
@@ -686,6 +686,18 @@ - (BOOL)isStateChangeValid:(ASVideoNodePlayerState)state
   return YES;
 }
 
+- (void)reset {
+  ASDN::MutexLocker l(__instanceLock__);
+  
+  if (_playerNode != nil) {
+    [_playerNode removeFromSupernode];
+    _playerNode = nil;
+  }
+  
+  [_player seekToTime:kCMTimeZero];
```

Comment:
Hi @appleguy I gave this a try and it turns out it's a bit more complicated than it at first seems. The changes required to support this would actually be quite a lot larger to get things to the point where you could re-start the playing of the video after the reset.

It turns out that the construction of the AVPlayer is tied up in `(void)prepareToPlayAsset:(AVAsset *)asset withKeys:(NSArray<NSString *> *)requestedKeys`. This is mixed in with the construction of the AVPlayerItem (which we'd actually want to keep hanging around) and setting up the delegates. I'd need to split prepareToPlayAsset up to separate the handling of the AVPlayer from the rest of the construction. 

I could have a go at this but I think it's best dealt with as a separate bug, keeping this PR with the seekToTime as above. I've actually been using the code above for a while in a live app and it seems to work fine, so I'm confident in the implementation.

Please let me know what you think.